src/libostree/ostree-repo-commit.c

```diff
@@ -424,6 +424,134 @@ add_size_index_to_metadata (OstreeRepo        *self,
 }
 
 static gboolean
+ostree_repo_commit_unpack_sizes (GVariant                    *entry,
+                                 OstreeContentSizeCacheEntry *sizes,
+                                 char                        *csum)
+{
+  gboolean ret = FALSE;
+  const guchar *buffer;
+  gsize bytes_read = 0;
+  gsize object_size = g_variant_get_size (entry);
+
+  if (object_size <= 32)
+    goto out;
+
+  buffer = g_variant_get_data (entry);
+  if (!buffer)
+    goto out;
+
+  ostree_checksum_inplace_from_bytes (buffer, csum);
+  buffer += 32;
+  object_size -= 32;
+
+  if (!_ostree_read_varuint64 (buffer, object_size, &(sizes->archived), &bytes_read))
+    goto out;
+  buffer += bytes_read;
+  object_size -= bytes_read;
+
+  if (!_ostree_read_varuint64 (buffer, object_size, &(sizes->unpacked), &bytes_read))
+    goto out;
+  buffer += bytes_read;
+  object_size -= bytes_read;
+
+  if (object_size < 1)
+    goto out;
+
+  if (*buffer < OSTREE_OBJECT_TYPE_FILE || *buffer > OSTREE_OBJECT_TYPE_LAST)
+    goto out;
```

Comment:
I think I (or John) was trying to stick with ostree style where there's very rarely early returns. This could _could_ go upstream (https://bugzilla.gnome.org/show_bug.cgi?id=756266), so I'd defer to upstream style.